atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -110,60 +114,79 @@ public void run() {
             Thread.sleep(getBackoffTimeWhenSweepHasNotRun());
             log.info("Starting background sweeper.");
             while (true) {
-                long millisToSleep = checkConfigAndRunSweep(locks);
-                Thread.sleep(millisToSleep);
+                SweepOutcome outcome = checkConfigAndRunSweep(locks);
+
+                updateBatchSize(outcome);
+                updateMetrics(outcome);
+
+                sleepUntilNextRun(outcome);
             }
         } catch (InterruptedException e) {
             log.warn("Shutting down background sweeper. Please restart the service to rerun background sweep.");
         }
     }
 
-    // Returns milliseconds to sleep
+    private void updateBatchSize(SweepOutcome outcome) {
+        if (outcome == SweepOutcome.SUCCESS) {
+            batchSizeMultiplier = Math.min(1.0, batchSizeMultiplier * 1.01);
+            return;
+        }
+        if (outcome == SweepOutcome.RETRYING_WITH_SMALLER_BATCH) {
+            SweepBatchConfig lastBatchConfig = specificTableSweeper.getAdjustedBatchConfig();
```

Comment:
This is named confusingly, this is `newBatchConfig`?

Reply:
This is kind of funky.  `getSweepBatchConfig` gets the 'base' config values.  `getAdjustedBatchConfig` get the `SweepBatchConfig` but adjusted by applying the `batchSizeMultiplier`.  It doesn't change the underlying config though.

This isn't the `newBatchConfig` though.  This is what we believe the config we ran with for the last batch was.  We haven't yet changed the `batchSizeMultiplier`.  This may be wrong though if the batch config is refreshed in the background.

Basically, we're not passing the actual config we used around, so this line is trying to recreate what was just used.

There's probably a cleaner way of doing this.